src/libcore/pin.rs

```diff
@@ -349,6 +349,20 @@ where
         // around pinning.
         unsafe { Pin::new_unchecked(pointer) }
     }
+
+    /// Unwraps this `Pin<P>` returning the underlying pointer to some data of a
+    /// type that implements [`Unpin`].
+    ///
+    /// Unlike `Pin::into_inner_unchecked`, this method is safe because the
+    /// pointer `P` dereferences to an [`Unpin`] type, which cancels the pinning
+    /// guarantees.
+    ///
+    /// [`Unpin`]: ../../std/marker/trait.Unpin.html
+    #[unstable(feature = "pin_into_inner", issue = "0")]
+    #[inline(always)]
+    pub fn into_inner(pin: Pin<P>) -> P {
```

Comment:
I'd expected we'd call this method `unpin`, though maybe that's too clever.

Reply:
Either seems fine to me, let me know if I should change it.